### PR TITLE
Add missing gRPC calls

### LINKED ISSUE #143: [Bug] DeleteWorkflowExecution service call not exposed properly
### Describe the bug

On the Rust side, we don't expose the `DeleteWorkflowExecution` call. Not only do we need to, but we need a unit test to make sure no calls are ever missed going forward. This should basically be us standing up a fake gRPC server, calling every call on workflow and operator service reflectively, and confirming every call reaches the server.

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

Added missing gRPC calls from the core string list and added test to ensure they are all there

## Checklist

1. Closes #143

### CODE REVIEW

File: tests/Temporalio.Tests/Client/TemporalClientTests.cs [hunk 1]
```diff
@@ -48,4 +52,124 @@ public async Task CreateLazy_Connection_NotConnectedUntilCallMade()
...
+        actualCalls.Clear();
+
+        // Make calls and populate expected calls
+        var expectedCalls = new List<string>();
+        foreach (var method in typeof(T).GetMethods(
+            BindingFlags.Instance | BindingFlags.Public | BindingFlags.DeclaredOnly))
+        {
+            // Add expected call sans Async suffix
+            var call = method.Name.Substring(0, method.Name.Length - 5);
+            Console.WriteLine("!!! CALL: " + call);
```
Comment (bergundy) [inline tests/Temporalio.Tests/Client/TemporalClientTests.cs]: Did you mean to leave this console statement here?

File: tests/Temporalio.Tests/Client/TemporalClientTests.cs [hunk 2]
```diff
@@ -48,4 +52,124 @@ public async Task CreateLazy_Connection_NotConnectedUntilCallMade()
...
+        {
+            Runtime = new(new()
+            {
+                Telemetry = new() { Metrics = new() { CustomMetricMeter = captureMeter } },
+            }),
+        });
+
+        // Check workflow service and operator service
+        await AssertAllRpcsAsync(captureMeter.Calls, env.Client.Connection.WorkflowService);
+        await AssertAllRpcsAsync(captureMeter.Calls, env.Client.Connection.OperatorService, skip: "AddOrUpdateRemoteCluster");
```
Comment (bergundy) [inline tests/Temporalio.Tests/Client/TemporalClientTests.cs:74]: Curious why this call is skipped and if this is temporary.

**Review by bergundy — APPROVED** [review]

File: tests/Temporalio.Tests/Client/TemporalClientTests.cs (hunk 1 above)
Comment (cretz) [reply tests/Temporalio.Tests/Client/TemporalClientTests.cs]: I did not, thanks!

File: tests/Temporalio.Tests/Client/TemporalClientTests.cs (hunk 2 above)
Comment (cretz) [reply tests/Temporalio.Tests/Client/TemporalClientTests.cs:74]: It hangs/messes up the server when you call it with an empty request. Not temporary, but hopefully it's the only one like it.

File: tests/Temporalio.Tests/Client/TemporalClientTests.cs (hunk 1 above)
Comment (cretz) [reply tests/Temporalio.Tests/Client/TemporalClientTests.cs]: Fixed

**Pull request merged**